src/groups/mqb/mqbnet/mqbnet_tcpsessionfactory.cpp

```diff
@@ -1414,5 +1409,34 @@ bool TCPSessionFactory::isEndpointLoopback(const bslstl::StringRef& uri) const
            mwcio::ChannelUtil::isLocalHost(endpoint.host());
 }
 
+// --------------------------------------
+// class TCPSessionFactory::PortExtractor
+// --------------------------------------
+
+TCPSessionFactory::PortExtractor::PortExtractor(bslma::Allocator* allocator)
+: d_regex(allocator)
+{
+    const char                  pattern[] = ":(\\d{1,5})$";
+    bsl::string                 error(allocator);
+    size_t                      errorOffset;
+    BSLA_MAYBE_UNUSED const int rc = d_regex.prepare(
+        &error,
+        &errorOffset,
+        pattern,
+        bdlpcre::RegEx::k_FLAG_JIT);
+    BSLS_ASSERT_SAFE(rc == 0);
+    BSLS_ASSERT_SAFE(d_regex.isPrepared() == true);
+}
+
+bsl::string_view
+TCPSessionFactory::PortExtractor::extract(const bsl::string& endpoint) const
+{
+    bsl::string_view result;
+
+    BSLS_ASSERT_SAFE(0 == d_regex.match(&result, endpoint));
+
+    return result.substr(1);
```

Comment:
Do we make `substr(1)` here because we capture the leading `:`?

Reply:
Yes, definetley. If we used the `match()` overload that puts results to a vector, we would just take the [1] element of the vector, that contains sub expresion of the regex, and the [0] element would contain the full match with `:` in the beginning. However current `match()` overload only returns full matched expression.